### PR TITLE
examples: Supabase | ui.shadcn, App Router Layouts

### PR DESCRIPTION
**Pull request by MildTomato**
- Updated to use App Router layouts
- Updated to use ui.shadcn
- Updated Forms to return errors from actions
- Bumped a few version numbers
- added next-themes
- added actions.ts
- user JSON now shown on login

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: c0b02c2e35145974bb882ad2b77dfd78a703856f

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: c0b02c2e35145974bb882ad2b77dfd78a703856f

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: examples/with-supabase/app/(auth-pages)/forgot-password/page.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { FormMessage, Message } from "@/components/form-message";
+import { headers } from "next/headers";
+import { encodedRedirect } from "@/utils/utils";
+
+export default function ForgotPassword({
+  searchParams,
+}: {
+  searchParams: Message;
+}) {
+  const forgotPassword = async (formData: FormData) => {
```
Comment (leerob) [inline examples/with-supabase/app/(auth-pages)/forgot-password/page.tsx:16]: Could we move this to an `actions.ts` file?

File: examples/with-supabase/app/(auth-pages)/login/page.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import { SubmitButton } from "@/components/submit-button";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { createClient } from "@/utils/supabase/server";
+import { encodedRedirect } from "@/utils/utils";
+import Link from "next/link";
+import { redirect } from "next/navigation";
+
+export default function Login({ searchParams }: { searchParams: Message }) {
+  const signIn = async (formData: FormData) => {
```
Comment (leerob) [inline examples/with-supabase/app/(auth-pages)/login/page.tsx:11]: Same

File: examples/with-supabase/app/(auth-pages)/signup/page.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import { headers } from "next/headers";
+import Link from "next/link";
+import { SubmitButton } from "@/components/submit-button";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { FormMessage, Message } from "@/components/form-message";
+import { encodedRedirect } from "@/utils/utils";
+
+export default function Signup({ searchParams }: { searchParams: Message }) {
+  const signUp = async (formData: FormData) => {
```
Comment (leerob) [inline examples/with-supabase/app/(auth-pages)/signup/page.tsx:11]: Same 😄

File: examples/with-supabase/components/auth-button.tsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    }
+  };
+
+  const isSupabaseConnected = canInitSupabaseClient();
+
+  const {
+    data: { user },
+  } = await supabase.auth.getUser();
+
+  const signOut = async () => {
```
Comment (leerob) [inline examples/with-supabase/components/auth-button.tsx:27]: Same 😄

File: examples/with-supabase/components/auth-button.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    "use server";
+
+    const supabase = createClient();
+    await supabase.auth.signOut();
+    return redirect("/login");
+  };
+
+  if (!isSupabaseConnected) {
+    // if (true) {
```
Comment (leerob) [inline examples/with-supabase/components/auth-button.tsx:36]: Remove?

File: examples/with-supabase/components/deploy-dropdown.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            <span>Deploy to Vercel</span>
+          </Button>
+        </DropdownMenuTrigger>
+        <DropdownMenuContent align="start" className="max-w-xs">
+          <DropdownMenuItem className="flex flex-col gap-1 items-start" asChild>
+            <Link
+              href="https://vercel.com/new/clone?repository-url=https%3A%2F%2Fgithub.com%2Fvercel%2Fnext.js%2Ftree%2Fcanary%2Fexamples%2Fwith-supabase&project-name=nextjs-with-supabase&repository-name=nextjs-with-supabase&demo-title=nextjs-with-supabase&demo-description=This%20starter%20configures%20Supabase%20Auth%20to%20use%20cookies%2C%20making%20the%20user's%20session%20available%20throughout%20the%20entire%20Next.js%20app%20-%20Client%20Components%2C%20Server%20Components%2C%20Route%20Handlers%2C%20Server%20Actions%20and%20Middleware.&demo-url=https%3A%2F%2Fdemo-nextjs-with-supabase.vercel.app%2F&external-id=https%3A%2F%2Fgithub.com%2Fvercel%2Fnext.js%2Ftree%2Fcanary%2Fexamples%2Fwith-supabase&demo-image=https%3A%2F%2Fdemo-nextjs-with-supabase.vercel.app%2Fopengraph-image.png&stores=%5B%7B%22type%22%3A%22integration%22%2C%22integrationSlug%22%3A%22supabase%22%2C%22productSlug%22%3A%22supabase%22%7D%5D"
```
Comment (leerob) [inline examples/with-supabase/components/deploy-dropdown.tsx:35]: Let's just to the simple button without the dropdown - just the first link.

File: examples/with-supabase/app/(auth-pages)/signup/page.tsx (hunk 3 above)
Comment (MildTomato) [reply examples/with-supabase/app/(auth-pages)/signup/page.tsx:11]: 😆

File: examples/with-supabase/components/auth-button.tsx (hunk 5 above)
Comment (MildTomato) [reply examples/with-supabase/components/auth-button.tsx:36]: Thanks.

File: examples/with-supabase/app/(auth-pages)/forgot-password/page.tsx (hunk 1 above)
Comment (leerob) [reply examples/with-supabase/app/(auth-pages)/forgot-password/page.tsx:16]: Best practice we recommend is to have in a dedicated file 🙏

File: examples/with-supabase/components/deploy-dropdown.tsx (hunk 6 above)
Comment (MildTomato) [reply examples/with-supabase/components/deploy-dropdown.tsx]: done!

File: examples/with-supabase/.env.example [hunk 7]
```diff
@@ -1,4 +0,0 @@
-# Update these with your Supabase details from your project settings > API
```
Comment (saltcod) [inline examples/with-supabase/.env.example:1]: was this deleted on purpose?
or just renamed & the .env.local isn't tracked so this looks deleted?

Comment (MildTomato) [reply examples/with-supabase/.env.example:1]: was a mistake

**Comment by MildTomato** [comment]
Still sorting some issues with instructions

File: examples/with-supabase/app/actions.ts [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+    return encodedRedirect(
+      "success",
+      "/signup",
+      "Thanks for signing up! Please check your email for a verification link.",
+    );
+  }
+};
+
+export const signInAction = async (formData: FormData) => {
+  "use server";
```
Comment (saltcod) [inline examples/with-supabase/app/actions.ts:39]: just need one 'use server' at the top?

Comment (MildTomato) [reply examples/with-supabase/app/actions.ts:39]: yea

File: examples/with-supabase/components/ui/badge.tsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import * as React from "react";
```
Comment (ivasilov) [inline examples/with-supabase/components/ui/badge.tsx:1]: All shadcn files are in `components/ui` but they should be in `components` folder according to the `components.json`. Change `components.json` (probably more correct) or move them back.

Comment (MildTomato) [reply examples/with-supabase/components/ui/badge.tsx:1]: shadcn adds `ui` directory itself

Comment (MildTomato) [reply examples/with-supabase/components/ui/badge.tsx:1]: its just not well documented

Comment (MildTomato) [reply examples/with-supabase/components/ui/badge.tsx:1]: this components.json file is what placed them in the right directory, i used the CLI to add components

**Review by leerob — APPROVED** [review]
Looking good!

**Pull request merged**